[test_sampling_1d.py] | (T3K) T3000 e2e tests | t3000-e2e-tests / models_tttv2_sampling_modules_full_set_unit_tests [wh_llmbox]

## CI Failure Report

<!-- AUTO-TRIAGE-SUMMARY-START -->
**Summary:** [CI] TT_FATAL: num_intersections == num_cores in test_sampling1d_padded_tail_with_sub_core_grids_runs[1x8] (T3000 sampling unit tests)
<!-- AUTO-TRIAGE-SUMMARY-END -->

**Workflow:** (T3K) T3000 e2e tests
**Job:** `t3000-e2e-tests / models_tttv2_sampling_modules_full_set_unit_tests [wh_llmbox]`

### Regression timeline

- **Last passing commit:** N/A (not provided)
- **First failing commit:** `7133403b29d9` — 2026-07-01T07:45:02Z — [job link](https://github.com/tenstorrent/tt-metal/actions/runs/28501910684/job/84486280668)
- **Most recent failing commit:** `fe02a0721bb7` — 2026-07-02T07:19:33Z — [job link](https://github.com/tenstorrent/tt-metal/actions/runs/28572645035/job/84716447206)

### Failing job URLs (last 2 runs)

- https://github.com/tenstorrent/tt-metal/actions/runs/28572645035/job/84716447206
- https://github.com/tenstorrent/tt-metal/actions/runs/28501910684/job/84486280668

### Failing test path(s)

- `models/common/tests/modules/sampling/test_sampling_1d.py::test_sampling1d_padded_tail_with_sub_core_grids_runs[1x8]`

**Test command:**
```
HF_HUB_OFFLINE=1 HF_HOME=/mnt/MLPerf/huggingface pytest --tb=short models/common/tests/modules/sampling
```

### Error excerpt (terminal failure from logs)

**Run 1** (commit `fe02a0721bb7`, 2026-07-02):
```
FAILED models/common/tests/modules/sampling/test_sampling_1d.py::test_sampling1d_padded_tail_with_sub_core_grids_runs[1x8] - RuntimeError: TT_FATAL @ /project/tt_metal/impl/program/program.cpp:1988: num_intersections == num_cores
Kernel group cores do not match sub device cores for programmable core type TENSIX
========== 1 failed, 533 passed, 694 deselected in 418.45s (0:06:58) ===========
```

**Run 2** (commit `7133403b29d9`, 2026-07-01):
```
FAILED models/common/tests/modules/sampling/test_sampling_1d.py::test_sampling1d_padded_tail_with_sub_core_grids_runs[1x8] - RuntimeError: TT_FATAL @ /project/tt_metal/impl/program/program.cpp:1988: num_intersections == num_cores
Kernel group cores do not match sub device cores for programmable core type TENSIX
========== 1 failed, 533 passed, 694 deselected in 411.32s (0:06:51) ===========
```

**Stack trace (both runs identical):**
```
models/common/tests/modules/sampling/test_sampling_1d.py:751: in test_sampling1d_padded_tail_with_sub_core_grids_runs
    tokens_tt, _ = sampler.decode_forward(logits_tt, k=k, p=p, temp=temp)
models/common/modules/sampling/sampling_1d.py:268: in decode_forward
    return self._sample_topk(logits, k, p, temp, seeds, tt_out_tok)
models/common/modules/sampling/sampling_1d.py:388: in _sample_topk
    topk_global_indices = ttnn.add(
/opt/venv/lib/python3.10/site-packages/ttnn/decorators.py:650: in __call__
    result = self.function(*function_args, **function_kwargs)
E   RuntimeError: TT_FATAL @ /project/tt_metal/impl/program/program.cpp:1988: num_intersections == num_cores
E   info:
E   Kernel group cores do not match sub device cores for programmable core type TENSIX
E    --- tt::tt_metal::detail::ProgramImpl::determine_sub_device_ids(tt::tt_metal::IDevice const*)
E    --- tt::tt_metal::detail::ProgramImpl::validate_circular_buffer_region(tt::tt_metal::IDevice const*)
E    --- tt::tt_metal::distributed::MeshWorkloadImpl::compile_program(...)
E    --- tt::tt_metal::distributed::MeshWorkloadImpl::compile(tt::tt_metal::distributed::MeshDevice*)
E    --- tt::tt_metal::distributed::EnqueueMeshWorkload(...)
```

### Reproduction steps

1. Set up a T3000 (wh_llmbox) environment with the tt-metal repo at commit `7133403b29d9` or later.
2. Run the sampling unit tests:
   ```bash
   HF_HUB_OFFLINE=1 HF_HOME=/mnt/MLPerf/huggingface pytest --tb=short models/common/tests/modules/sampling
   ```
3. Observe the failure in `test_sampling1d_padded_tail_with_sub_core_grids_runs[1x8]`.

### Notes

The assertion `num_intersections == num_cores` fires in `program.cpp:1988` during `ProgramImpl::determine_sub_device_ids()`. The error message states _"Kernel group cores do not match sub device cores for programmable core type TENSIX"_, indicating a mismatch between the core set assigned to a kernel group and the sub-device core set when running `ttnn.add` on padded-tail/sub-core-grid configuration `[1x8]`.

**Determinism check:**
- **Run 1** terminal error: `FAILED models/common/tests/modules/sampling/test_sampling_1d.py::test_sampling1d_padded_tail_with_sub_core_grids_runs[1x8] - RuntimeError: TT_FATAL @ /project/tt_metal/impl/program/program.cpp:1988: num_intersections == num_cores`
- **Run 2** terminal error: `FAILED models/common/tests/modules/sampling/test_sampling_1d.py::test_sampling1d_padded_tail_with_sub_core_grids_runs[1x8] - RuntimeError: TT_FATAL @ /project/tt_metal/impl/program/program.cpp:1988: num_intersections == num_cores`
- Both runs: 1 failed, 533 passed, 694 deselected — identical failure count and test case.

<!-- AUTO-TRIAGE-METADATA-START -->
`Auto-triage-workflow: (T3K) T3000 e2e tests`
`Auto-triage-job-name: t3000-e2e-tests / models_tttv2_sampling_modules_full_set_unit_tests [wh_llmbox]`
<!-- AUTO-TRIAGE-METADATA-END -->

<!-- AUTO-TRIAGE-INFO-START -->
## About this issue (automated CI auto-triage)
This issue was opened and assigned by an automated AI system. It posts through a bot token, so notifications may look like they come from a person, but they are automated.

The owner is selected from recent commit history and job-ownership data. It reflects who is *likely able to fix or route* this, not necessarily who caused it, so it can be imperfect.

**If this was assigned to the wrong person, you can:**
- Comment `REASSIGN OWNER: <why this is outside your area / which team or test-area it belongs to>` and the system will automatically re-route it to a better-suited owner.
- Unassign yourself (this stops notifications to you).
- Close the issue if it is not a real or relevant failure.

The assigned owner is reminded about every 3 days until the issue is resolved. Full guide: https://tenstorrent.atlassian.net/wiki/spaces/MI6/pages/2545221692/Auto-Triage+Info
<!-- AUTO-TRIAGE-INFO-END -->

--- a/models/common/modules/sampling/sampling_1d.py
+++ b/models/common/modules/sampling/sampling_1d.py
@@ -390,6 +390,7 @@ class Sampling1D(LightweightModule):
             topk_indices_int32,
             dtype=ttnn.int32,
             memory_config=cfg.sampling_memory_config,
+            sub_core_grids=cfg.sub_core_grids,
         )
         ttnn.deallocate(topk_indices_int32)
 

